ipatests/test_integration/test_acme.py

```diff
@@ -542,3 +552,129 @@ def install(cls, mh):
+        self.master.run_command(['ipa-acme-manage', 'enable'])
+
+        # register the account with certbot
+        # clean up any existing registration and certificates
+        self.clients[0].run_command(
+            [
+                'rm', '-rf',
+                '/etc/letsencrypt/accounts',
+                '/etc/letsencrypt/archive',
+                '/etc/letsencrypt/csr',
+                '/etc/letsencrypt/keys',
+                '/etc/letsencrypt/live',
+                '/etc/letsencrypt/renewal',
+                '/etc/letsencrypt/renewal-hooks'
+            ]
+        )
+        # service is enabled; registration should succeed
+        self.clients[0].run_command(
+            [
+                'certbot',
+                '--server', self.acme_server,
+                'register',
+                '-m', 'nobody@example.test',
+                '--agree-tos',
+                '--no-eff-email',
+            ],
+        )
+
+        # request a standalone acme cert
+        self.clients[0].run_command(['systemctl', 'stop', 'httpd'])
+        self.clients[0].run_command(
+            [
+                'certbot',
+                '--server', self.acme_server,
+                'certonly',
+                '--domain', self.clients[0].hostname,
+                '--standalone',
+            ],
+        )
+
+        cmd_input = (
+            # Password for admin@{REALM}:
+            "{pwd}\n"
+            # Password expired.  You must change it now.
+            # Enter new password:
+            "{pwd}\n"
+            # Enter it again:
+            "{pwd}\n"
+        )
+        # move system date to expire acme cert
+        for host in self.clients[0], self.master:
+            host.run_command(['kdestroy', '-A'])
+            move_date(host, 'stop', '+90days')
+            host.run_command(
+                ['kinit', 'admin'],
+                stdin_text=cmd_input.format(pwd=host.config.admin_password)
+            )
+
+        yield
+
+        # move back date
+        for host in self.clients[0], self.master:
+            host.run_command(['kdestroy', '-A'])
+            move_date(host, 'start', '-90days')
+            host.run_command(
+                ['kinit', 'admin'],
+                stdin_text=cmd_input.format(pwd=host.config.admin_password)
+            )
+
+    def test_renew(self, issue_and_expire_cert):
+        """Test if ACME renews the issued cert with cerbot
+
+        This test is to check if ACME certificate renews upon
+        reaching expiry
+
+        related: https://pagure.io/freeipa/issue/4751
+        """
+        data = self.clients[0].get_file_contents(
+            f'/etc/letsencrypt/live/{self.clients[0].hostname}/cert.pem'
+        )
+        cert = x509.load_pem_x509_certificate(data, backend=default_backend())
+        initial_expiry = cert.not_valid_after
+
+        tasks.kinit_admin(self.clients[0])
+
+        self.clients[0].run_command(['certbot', 'renew'])
+        time.sleep(10)
```

Comment:
Is the sleep really needed?

Reply:
yes, Just to be sure that all the bits get renewed.

Reply:
The help for certbot renew does not mention that the command could return before the cert actually gets renewed. Did you see issues when the sleep(10) was not present?